CesiumIonClient/include/CesiumIonClient/ApplicationData.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace CesiumIonClient {
+
+/**
+ *  @brief An enumeration representing the values of {@link ApplicationData::authenticationMode}.
+ */
+enum AuthenticationMode {
+  /**
+   * Authentication using OAuth with an ion.cesium.com account.
+   */
+  CesiumIon,
+  /**
+   * Authentication using OAuth with Cesium ion Self-Hosted.
+   * On the server, this uses the Security Assertion Markup Language (SAML) to
+   * communicate with another authentication server.
+   * From our perspective, we can treat this the same as {@link AuthenticationMode::CesiumIon}.
+   */
+  Saml,
+  /**
+   * A Cesium ion Self-Hosted server without authentication.
+   * In single-user mode, any application that can reach the server has
+   * permissions to use its endpoints. In this mode, some endpoints (like /me
+   * and /tokens) are unavailable.
+   */
+  SingleUser
+};
+
+/**
+ * @brief Data about the Cesium ion server itself.
+ */
+struct ApplicationData {
```

Comment:
@azrogers I'm realizing that I forgot to leave a comment on this :sob:

I think a better name would be `ServerData` instead? At least for me, `ApplicationData` implies data about the application that the server is being used for, and not the server itself.

I don't want to trouble you for another find + replace change, so I'll push a commit for it but please let me know if you disagree.

Reply:
eh actually on second thought, Cesium ion _does_ call it appData so I guess we should keep consistent. I'll just add documentation tweaks so it's a little more obvious